src/parser/shared/modules/StatTracker.js

```diff
@@ -522,6 +447,30 @@ class StatTracker extends Analyzer {
     debug && this._debugPrintStats(this._currentStats);
   }
 
+  /**
+   * Adds a stat buff to StatTracker.
+   * @param buffId ID of the stat buff
+   * @param stats Object with stats (intellect, mastery, haste, crit etc.) and their respective bonus (either fixed value or a function (combatant, item) => value). If it's from item, provide also an itemId (item in the stat callback is taken from this itemId).
+   */
+  add(buffId, stats) {
+    if (!buffId || !stats) {
+      debug && this.log(`StatTracker.add() called with invalid buffId ${buffId} or stats`, stats);
+      return;
+    }
+    if (this.constructor.STAT_BUFFS[buffId]) {
+      debug && this.log(`Stat buff with ID ${buffId} already exists`);
```

Comment:
I think in dev mode these should probably throw an error rather than just logging -- just a thought